test/runtime/ct.go

```diff
@@ -0,0 +1,438 @@
+
+		for _, testCase := range testCombinations {
+			//Test policies are still applied correctly
+			assertfn := BeTrue
+			if testCase.destination[helpers.Name] == server2 {
+				assertfn = BeFalse
+			}
+			testReach(testCase.src[helpers.Name], testCase.destination[testCase.kind], testCase.mode, assertfn)
+		}
+	})
+
+	It("test the L7 CT cleanup", func() {
+		vm.SetPolicyEnforcement(helpers.PolicyEnforcementAlways)
+		containers(helpers.Create)
+		defer containers(helpers.Delete)
+		vm.WaitEndpointsReady()
+
+		meta := containersMeta()
+		identities, err := vm.GetEndpointsIdentityIds()
+
+		By("Checking Policy L4")
+		Expect(err).To(BeNil())
+		policyL4 := `
+		[{
+			"endpointSelector": {"matchLabels":{"id.server-3":""}},
+			"ingress": [{
+				"fromEndpoints": [{
+				   "matchLabels":{"id.client":""}
+				}],
+				"toPorts": [{
+					"ports": [{"port": "80", "protocol": "tcp"}]
+				}]
+			}],
+			"labels": ["id=server-3"]
+		}]
+		`
+		_, err = vm.PolicyRenderAndImport(policyL4)
+		Expect(err).To(BeNil())
+
+		testCombinations := []connTest{
+			{meta[client], meta[server3], helpers.IPv6, HTTPPrivate},
+			{meta[client], meta[server3], helpers.IPv4, HTTPPrivate},
+			{meta[client], meta[server3], helpers.IPv6, HTTPPublic},
+			{meta[client], meta[server3], helpers.IPv4, HTTPPublic},
+			{meta[client], meta[server3], helpers.IPv6, netcatPrivate},
+			{meta[client], meta[server3], helpers.IPv4, netcatPrivate},
+			{meta[client], meta[server3], helpers.IPv6, netcatPublic},
+			{meta[client], meta[server3], helpers.IPv4, netcatPublic},
+		}
+
+		for _, testCase := range testCombinations {
+			testReach(testCase.src[helpers.Name], testCase.destination[testCase.kind], testCase.mode, BeTrue)
+		}
+
+		res := vm.PolicyDel("id=server-3")
+		res.WasSuccessful()
+
+		By("Checking Policy L7")
+		policyL7 := `
+		[{
+			"endpointSelector": {"matchLabels":{"id.server-3":""}},
+			"ingress": [{
+				"fromEndpoints": [{
+				   "matchLabels":{"id.client":""}
+				}],
+				"toPorts": [{
+					"ports": [{"port": "80", "protocol": "tcp"}],
+					"rules": {"http": [{
+						  "path": "/public",
+						  "method": "GET"
+					}]}
+				}]
+			}],
+			"labels": ["id=server-3"]
+		}]`
+		_, err = vm.PolicyRenderAndImport(policyL7)
+		Expect(err).To(BeNil())
+
+		vm.ExecCilium("bpf ct flush global").WasSuccessful()
+		CTBefore := map[string]int{}
+		for _, testCase := range testCombinations {
+			assertfn := BeTrue
+			if strings.Contains(strings.ToLower(testCase.mode), "private") {
+				assertfn = BeFalse
+			}
+			testReach(testCase.src[helpers.Name], testCase.destination[testCase.kind], testCase.mode, assertfn)
+			data, err := countCTEntriesof(
+				testCase.destination[testCase.kind],
+				testCase.src[testCase.kind],
+				identities[testCase.src[helpers.Name]])
+			Expect(err).To(BeNil())
+			CTBefore[testCase.String()] = data
+		}
+
+		for k, v := range CTBefore {
+			Expect(v).To(Equal(6), "CT Tables are not correct for %s", k)
```

Comment:
I don't understand the message `"CT Tables are not correct for %s"`, what did you intend with this?